awsx/ec2/subnetDistributor.ts

```diff
@@ -98,7 +100,7 @@ export function getSubnetSpecs(
         azName: azNames[i],
         cidrBlock: publicSubnetCidrBlock,
         type: "Public",
-        subnetName: `${vpcName}-${publicSubnetsIn[j].name ?? "public"}-${i + 1}`,
+        subnetName: `${vpcName}-${publicSubnetsIn[j].name ?? "public"}-${azSuffix}`,
         tags: publicSubnetsIn[j].tags,
       });
```

Comment:
Let's add aliases here as you suggested to maintain backward compatibility with the old names.